pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -427,3 +439,34 @@ func UpdateCephDeploymentAndWait(context *clusterd.Context, clusterInfo *client.
 	err := k8sutil.UpdateDeploymentAndWait(clusterInfo.Context, context, deployment, clusterInfo.Namespace, callback)
 	return err
 }
+
+// floatingMonInitContainer configures the floating mon pod with an init container and volumes
+// from a ConfigMap named "rook-ceph-floating-mon-config"
+func (c *Cluster) floatingMonInitContainer(podSpec *corev1.PodSpec, monConfig *monConfig) error {
+	log.NamespacedDebug(c.Namespace, logger, "fetching configmap for floating mon %q init-container", monConfig.DaemonName)
+	floatingMonCm := "rook-ceph-floating-mon-config"
+	cm, err := c.context.Clientset.CoreV1().ConfigMaps(c.Namespace).Get(c.ClusterInfo.Context, floatingMonCm, metav1.GetOptions{})
+	if err != nil {
+		logger.Errorf("failed to get floating mon configmap %q. %v", floatingMonCm, err)
+		return nil
+	}
+
+	log.NamespacedDebug(c.Namespace, logger, "configuring floating mon pod %q init-container", monConfig.DaemonName)
+
+	var initContainer corev1.Container
+	err = yaml.Unmarshal([]byte(cm.Data["init-container.yaml"]), &initContainer)
```

Comment:
We should not use this method to update the mon deployments, We should write it in the golang as we wrote for other containers,

Similar way we don't want the volumes and volume mounts to be like that.

If we externally manage these dependency it will become un-manageble after certain time, when we need common some changes for al conatiners or common setting changes.